torch/_inductor/scheduler.py

```diff
@@ -628,11 +628,18 @@ def used_buffer_names(self) -> OrderedSet[str]:
         )
 
     def used_or_aliased_buffer_names(self) -> OrderedSet[str]:
+        """
+        Returns buffer names used by this node, including aliases.
+
+        Note: is_fake WeakDeps are excluded since they are purely for ordering
+        and should not affect buffer lifetime.
+        """
         used_names: OrderedSet[str] = OrderedSet()
 
         deps = [
             dep.name
             for dep in itertools.chain(self.read_writes.reads, self.read_writes.writes)
+            if not (isinstance(dep, WeakDep) and dep.is_fake)
```

Comment:
worth a API in depencencies.py to check for fake dep since it's used in multiple places?